File: mirai/edge/template/mirai_tools/csharp/MiraiSetup.cs
using Mirai;

/// <summary>
/// Mirai edge setup — register your tools here.
/// Call <see cref="MiraiSetup.InitMirai"/> from your application's entry point.
/// </summary>
public static class MiraiSetup
{
    // ── Configuration ──
    // Set your connection code here (LAN: "mirai-lan_...", remote: "mirai_...").
    // Leave empty to read from MIRAI_CONNECTION_CODE env var or mirai_tools/.env.
    private const string MiraiConnectionCode = "";
    private const string MiraiEdgeName = "My C# App";

    public static MiraiAgent InitMirai()
    {
        var agent = new MiraiAgent(MiraiConnectionCode, MiraiEdgeName);

        // Register your tools below.

        agent.Register(new RegisterOptions()
            .SetName("hello")
            .SetDescription("Say hello to someone")
            .SetParameters(
                new ToolParameter("name", "string", "Person to greet")
            )
            .SetHandler(args =>
            {
                var name = args.GetString("name", "World");
                return $"Hello, {name}!";
            })
        );

        // Example: tool with confirmation required
        // agent.Register(new RegisterOptions()
        //     .SetName("dangerous_action")
        //     .SetDescription("Do something irreversible")
        //     .SetRequireConfirmation(true)
        //     .SetHandler(args => "done")
        // );

        agent.RunInBackground();
        return agent;
    }
}

File: mirai/edge/template/mirai_tools/java/MiraiSetup.java
import io.mirai.MiraiAgent;
import io.mirai.RegisterOptions;
import io.mirai.ToolParameter;

/**
 * Mirai edge setup — register your tools here.
 * Call {@link #initMirai()} from your application's main method.
 */
public class MiraiSetup {

    // ── Configuration ──
    // Set your connection code here (LAN: "mirai-lan_...", remote: "mirai_...").
    // Leave empty to read from MIRAI_CONNECTION_CODE env var or mirai_tools/.env.
    private static final String MIRAI_CONNECTION_CODE = "";
    private static final String MIRAI_EDGE_NAME = "My Java App";

    public static void initMirai() {
        var agent = new MiraiAgent(MIRAI_CONNECTION_CODE, MIRAI_EDGE_NAME);

        // Register your tools below.

        agent.register(new RegisterOptions()
            .name("hello")
            .description("Say hello to someone")
            .parameters(
                new ToolParameter("name", "string", "Person to greet")
            )
            .handler(args -> {
                String name = args.getString("name", "World");
                return "Hello, " + name + "!";
            })
        );

        // Example: tool with confirmation required
        // agent.register(new RegisterOptions()
        //     .name("dangerous_action")
        //     .description("Do something irreversible")
        //     .requireConfirmation(true)
        //     .handler(args -> "done")
        // );

        agent.runInBackground();
    }
}
